src/crimson/common/config_proxy.h

```diff
@@ -152,9 +155,19 @@ class ConfigProxy : public seastar::peering_sharded_service<ConfigProxy>
       }
     });
   }
+
   int get_val(std::string_view key, std::string *val) const {
     return get_config().get_val(*values, key, val);
   }
+
+  void set_val_default(const std::string& key, const std::string& val) {
+    get_config().set_val_default(*values, obs_mgr, key, val);
+  }
```

Comment:
This breaks the current interface logic of crimson's ConfigProxy.
See how `set_val` is implemented. `set_val_default` should use do_change for the reason stated in the above comments.
 ```
  seastar::future<> set_val_default(const std::string& key,
			    const std::string& val) {
    ceph::global::g_conf_set_val(key, val);
    return do_change([key, val, this](ConfigValues& values) {
      std::stringstream err;
      auto ret = get_config().set_val_default(values, obs_mgr, key, val, &err);
      if (ret < 0) {
	throw std::invalid_argument(err.str());
      }
    });
  }
```
Otherwise, the changes won't apply to all shards and the observers won't be notified.

Reply:
I have updated a comment, the function is required to apply the change on a individual shard.